hack/make-rules/deploy.sh

```diff
@@ -19,43 +19,31 @@ set -o errexit -o nounset -o pipefail
 REPO_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/../.." && pwd -P)"
 cd "${REPO_ROOT}"
 
+# make sure we have a k8s.io clone for the prod terraform
+k8sio_dir="$(cd "${REPO_ROOT}"/../k8s.io && pwd -P)"
+if [[ ! -d "${k8sio_dir}" ]]; then
+    >&2 echo "Deploying requires a github.com/kubernetes/k8s.io clone at ./../k8s.io"
+    >&2 echo "FAIL"
+    exit 1
+fi
+
+# install crane and get current image digest
 TAG="${TAG:-"$(date +v%Y%m%d)-$(git describe --always --dirty)"}"
-SERVICE_BASENAME="${SERVICE_BASENAME:-k8s-infra-oci-proxy}"
+# TODO: this can't actually be overridden currently
+# the terraform always uses the default here
 IMAGE_REPO="${IMAGE_REPO:-gcr.io/k8s-staging-infra-tools/archeio}"
-PROJECT="${PROJECT:-k8s-infra-oci-proxy}"
+(cd "${REPO_ROOT}"/hack/tools && go build -o "${REPO_ROOT}"/bin/crane github.com/google/go-containerregistry/cmd/crane)
+IMAGE_DIGEST="${IMAGE_DIGEST:-$(bin/crane digest "${IMAGE_REPO}:${TAG}")}"
 
-REGIONS=(
-    asia-east1
-    asia-northeast1
-    asia-northeast2
-    asia-south1
-    australia-southeast1
-    europe-north1
-    europe-southwest1
-    europe-west1
-    europe-west2
-    europe-west4
-    europe-west8
-    europe-west9
-    southamerica-west1
-    us-central1
-    us-east1
-    us-east4
-    us-east5
-    us-south1
-    us-west1
-    us-west2
-)
+# cd to prod terraform and apply
+cd "${k8sio_dir}"/infra/gcp/terraform/k8s-infra-oci-proxy-prod
```

Comment:
why not sure `gcloud` to get any metadata about the CR services we run in prod ? the SA could have viewer access to the prod?

Reply:
This deploys to staging project only